addons/binding/org.openhab.binding.wink/src/main/java/org/openhab/binding/wink/handler/WinkHub2BridgeHandler.java

```diff
@@ -1,154 +1,215 @@
+	 */
+	public void unLockDevice(IWinkDevice device) {
+		Map<String, String> updatedState = new HashMap<String, String>();
+		updatedState.put("locked", "false");
+		this.setDesiredState(device, updatedState);
+	}
+
+	/**
+	 * Sets the dimmer level of a device to the level specified as a percentage
+	 *
+	 * @param device The device to change brightness
+	 * @param level The percentage of brightness desired
+	 */
+	public void setDeviceDimmerLevel(IWinkDevice device, int level) {
+		Map<String, String> updatedState = new HashMap<String, String>();
+		if (level > 0) {
+			Float fLevel = Float.valueOf(level);
+			updatedState.put("powered", "true");
+			updatedState.put("brightness", String.valueOf(fLevel / 100.0f));
+		} else {
+			updatedState.put("powered", "false");
+		}
+		this.setDesiredState(device, updatedState);
+	}
+
+	/**
+	 * Retrieve a specified device from the Hub API
+	 *
+	 * @param deviceType The type of device to retrieve
+	 * @param uuid The unique identifier for the device
+	 * @return The device
+	 */
+	public IWinkDevice getDevice(WinkSupportedDevice deviceType, String uuid) {
+		logger.debug("Getting device through handler {}", uuid);
+		return client.getDevice(deviceType, uuid);
+	}
+
+	/**
+	 * Sets the thermostat temperature to the level specified in celcius
+	 *
+	 * @param device The device to change temperature
+	 * @param temperature The temperature desired
+	 */
+	public void setThermostatTemperature(IWinkDevice device, float temperature) {
+		Map<String, String> updatedState = new HashMap<String, String>();
+		Map<String, String> jsonData = device.getCurrentStateComplexJson();
+
+		String units = "f";
+		if (jsonData.get("units") != null && !jsonData.get("units").equals("null")) {
+			units = jsonData.get("units");
+		}
+
+		float desiredTemperature = temperature;
+		if (units.equals("f")) {
+			// Convert to Celcius before setting new temp.
+			desiredTemperature = (desiredTemperature - 32) / 1.8f;
+		}
+
+		if (jsonData.get("mode") != null && !jsonData.get("mode").equals("null")) {
+			String currentOperationMode = jsonData.get("mode");
+			if (currentOperationMode.equals("cool_only")) {
+				updatedState.put("max_set_point", String.valueOf(desiredTemperature));
+			} else if (currentOperationMode.equals("heat_only")) {
+				updatedState.put("min_set_point", String.valueOf(desiredTemperature));
+			} else { // auto
+				// Set them both the same.
+				updatedState.put("min_set_point", String.valueOf(desiredTemperature));
+				updatedState.put("max_set_point", String.valueOf(desiredTemperature));
+			}
+
+			logger.debug("Setting new temperature to {}", desiredTemperature);
+			this.setDesiredState(device, updatedState);
+		}
+	}
+
+	/**
+	 * Sets the thermostat mode to the mode specified
+	 *
+	 * @param device The device to change temperature
+	 * @param mode The desired mode of operation, heat_only, cool_only, auto
+	 */
+	public void setThermostatMode(IWinkDevice device, String mode) {

```

Comment:
Same for this method.  Add the specifics of the device to the device handler and then use this class to just set the values on the hub.

Reply:
That makes sense.  I have moved the thermostat related methods.